conans/test/unittests/client/tools/cppstd_required_test.py

```diff
@@ -0,0 +1,183 @@
+import unittest
+from mock import mock
+from parameterized import parameterized
+
+from conans.test.utils.conanfile import MockConanfile, MockSettings
+from conans.client.tools import OSInfo
+from conans.errors import ConanInvalidConfiguration
+
+from conans.tools import check_min_cppstd, valid_min_cppstd
+
+
+class UserProneTests(unittest.TestCase):
+
+    def setUp(self):
+        self.conanfile = MockConanfile(MockSettings({}))
+
+    @parameterized.expand(["check_min_cppstd", "valid_min_cppstd"])
+    def test_check_none_cppstd(self, method):
+        """ Cppstd must use a valid number as described in settings.yml
+        """
+        with self.assertRaises(AssertionError) as asserts:
+            eval(method)(self.conanfile, None, False)
+        self.assertEqual("Cannot check invalid cppstd version", str(asserts.exception))
+
+    @parameterized.expand(["check_min_cppstd", "valid_min_cppstd"])
+    def test_check_none_conanfile(self, method):
+        """ conanfile must be a ConanFile object
+        """
+        with self.assertRaises(AssertionError) as raises:
+            eval(method)(None, "17", False)
+        self.assertEqual("conanfile must be a ConanFile object", str(raises.exception))
+
+
+class CheckMinCppStdTests(unittest.TestCase):
+
+    def setUp(self):
+        self.conanfile = self._create_conanfile("gcc", "9", "Linux", "17", "libcxx")
```

Comment:
Move this to each test function, it is just a line calling the helper function

Reply:
done